pycbc/filter/resample.py

```diff
@@ -31,6 +32,10 @@
 _resample_func = {numpy.dtype('float32'): lal.ResampleREAL4TimeSeries,
                  numpy.dtype('float64'): lal.ResampleREAL8TimeSeries}
 
+@functools.lru_cache(maxsize=20)
+def cached_firwin(*args, **kwargs):
+    return scipy.signal.firwin(*args, **kwargs)
```

Comment:
```suggestion
def cached_firwin(*args, **kwargs):
    """Cache the FIR filter coefficients.
    This is mostly done for PyCBC Live, which rapidly and repeatedly resamples data.
    """
    return scipy.signal.firwin(*args, **kwargs)
```

Reply:
While Live is the main "consumser" of this, it will also speed up the data conditioning for both PE and PyGRB, when multiple, equal length, data series are read in.